src/test/java/de/tum/in/www1/artemis/localvcci/LocalVCLocalCIIntegrationTest.java

```diff
@@ -500,9 +500,10 @@ void testFetchPush_assignmentRepository_examMode() throws Exception {
         examRepository.save(exam);
 
         // Create StudentExam.
-        StudentExam studentExam = examUtilService.addStudentExam(exam);
-        studentExam.setUser(student1);
+
+        StudentExam studentExam = examUtilService.addStudentExamWithUser(exam, student1);
         studentExam.setExercises(List.of(programmingExercise));
+        studentExam.setWorkingTime(null);
```

Comment:
Sooo since I've added `setWorkingTime` in `ExamUtilService.addStudentExam`, this test failed deep inside some checks. Setting it back to `null` fixes the test, however, it somehow does not feel right. Sb. has an Idea?

Reply:
I think this is because of the line `exam.setStartDate(ZonedDateTime.now().minusHours(1));` below. If you look into the helper method creating the exam, you can see that the working time is set to 3000 (seconds). So the working time is already over when it is tested that the student can fetch/push etc. If you align the start date and end date of the exam with the woking time this part should work.

I'm not sure if the working time has any effect after the due date has passed (last part of the test). It should be possible to avoid setting the working time to null.